envoy-control-source-consul/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/consul/services/ConsulServiceChanges.kt

```diff
@@ -111,20 +121,25 @@ class ConsulServiceChanges(
         }
 
         private fun handleServicesChange(services: RecipesServices) = synchronized(servicesLock) {
-            initialLoader.update(services.serviceNames())
+            val serviceNames = services.serviceNames()
+                .filterTo(HashSet()) { shouldBeWatched(it, services.tagsForServiceOrNull(it) ?: emptyList()) }
+            initialLoader.update(serviceNames)
 
-            val newServices = services.serviceNames() - lastServices
+            val newServices = serviceNames - lastServices
             newServices.forEach { service ->
                 handleNewService(service)
                 Thread.sleep(subscriptionDelay.toMillis())
             }
 
-            val removedServices = lastServices - services.serviceNames()
+            val removedServices = lastServices - serviceNames
             removedServices.forEach { handleServiceRemoval(it) }
 
-            lastServices = services.serviceNames()
+            lastServices = serviceNames
         }
 
+        private fun shouldBeWatched(service: String, tags: List<String>): Boolean =
```

Comment:
could u explain why explicit method to call service instead of just calling the service - which actually has the same signature and method name?

Reply:
My bad, it was suppose to handle nullable list of tags, like so:
```java
 serviceWatchPolicy.shouldBeWatched(service, tags ?: emptyList())
```

Reply:
let it be